### PR TITLE
Speed up decoding column id

### PR DESCRIPTION
**Pull request by ngaut**
### What problem does this PR solve? 
Speedup decoding column ID by removing a redundant varint decoding.

TikV doesn't have the issue. See https://github.com/tikv/tikv/blob/master/src/coprocessor/codec/table.rs#L374

### What is changed and how it works?
The decode varint process called twice, this PR make it once.

### CODE REVIEW

**Comment by ngaut** [comment]
PTAL @lysu

**Comment by shenli** [comment]
@ngaut The title and description are mal-formated.

File: util/codec/codec.go [hunk 1]
```diff
@@ -442,6 +452,28 @@ func SetRawValues(data []byte, values []types.Datum) error {
...
+	}
+	flag := b[0]
+	length++
+	b = b[1:]
+	var l int
+	switch flag {
+	case varintFlag:
+		n, l = binary.Varint(b)
+		if n < 0 {
+			return 0, 0, errors.New("value larger than 64 bits")
```
Comment (coocood) [inline util/codec/codec.go]: We need to check `l` instead of `n`.

File: util/codec/codec.go [hunk 2]
```diff
@@ -442,6 +452,28 @@ func SetRawValues(data []byte, values []types.Datum) error {
...
 }
 
+// decodeVInt decodes the first encoded value as a varint from b and returns varint and its length.
+func decodeVInt(b []byte) (length int, n int64, err error) {
+	if len(b) < 1 {
+		return 0, 0, errors.New("invalid encoded key")
+	}
+	flag := b[0]
+	length++
+	b = b[1:]
```
Comment (coocood) [inline util/codec/codec.go]: How about call `DecodeVarint` directly?
The length is `len(remainedBytes) - len(b)`.

**Pull request merged**